services/pkgreview/package-rating.service.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { starRating, metric } = require('../text-formatters')
+const { colorScale } = require('../color-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+
+const pkgReviewColor = colorScale([2, 3, 4])
+
+const schema = Joi.object({
+  rating: [
+    Joi.number()
+      .min(1)
+      .max(5)
+      .optional(),
+    Joi.allow(null),
+  ],
+  reviewsCount: nonNegativeInteger,
+}).required()
+
+module.exports = class PkgreviewRating extends BaseJsonService {
+  static get category() {
+    return 'rating'
+  }
+
+  static get route() {
+    return {
+      base: 'pkgreview',
+      pattern: ':format(rating|stars)/:pkgManager/:pkgSlug+',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'pkgreview.dev Package Ratings',
+        pattern: 'rating/:pkgManager/:pkgSlug+',
+        namedParams: { pkgManager: 'npm', pkgSlug: 'react' },
+        staticPreview: this.render({
+          format: 'rating',
+          rating: 0.7,
+          reviewsCount: 237,
+        }),
+      },
+      {
+        title: 'pkgreview.dev Star Ratings',
+        pattern: 'stars/:pkgManager/:pkgSlug+',
+        namedParams: { pkgManager: 'npm', pkgSlug: 'react' },
+        staticPreview: this.render({
+          format: 'stars',
+          rating: 0.3,
+          reviewsCount: 200,
+        }),
+      },
+    ]
+  }
+
+  static render({ rating, reviewsCount, format }) {
```

Comment:
I've added some additional items below for `transform` (namely to remove it) and `handle` to refactor things a bit along with simplifying this function (for example the `rating * 5` expression currently exists in three places within this function). Just make sure to update the example listing aboves to use the corresponding `rating` value (i.e. `0.7` -> `3.5`, `0.3` -> `1.5`)

```js
  static render({ rating, reviewsCount, format }) {
    const message =
      format === 'rating'
        ? `${rating.toFixed(1)}/5 (${metric(reviewsCount)})`
        : starRating(rating)

    return {
      message,
      label: format,
      color: floorCount(rating, 2, 3, 4),
    }
  }
```